source/common/stream_info/stream_info_impl.h

```diff
@@ -481,24 +494,25 @@ struct StreamInfoImpl : public StreamInfo {
   }
 
   std::shared_ptr<UpstreamInfo> upstream_info_;
-  uint64_t bytes_received_{};
-  uint64_t bytes_retransmitted_{};
-  uint64_t packets_retransmitted_{};
-  uint64_t bytes_sent_{};
   const Network::ConnectionInfoProviderSharedPtr downstream_connection_info_provider_;
   const Http::RequestHeaderMap* request_headers_{};
   StreamIdProviderSharedPtr stream_id_provider_;
   absl::optional<DownstreamTiming> downstream_timing_;
   absl::optional<Upstream::ClusterInfoConstSharedPtr> upstream_cluster_info_;
-  Tracing::Reason trace_reason_;
   // Default construct the object because upstream stream is not constructed in some cases.
   BytesMeterSharedPtr upstream_bytes_meter_{std::make_shared<BytesMeter>()};
   BytesMeterSharedPtr downstream_bytes_meter_;
-  bool is_shadow_{false};
   std::string downstream_transport_failure_reason_;
+  OptRef<const StreamInfo> parent_stream_info_;
+  uint64_t bytes_received_{};
+  uint64_t bytes_retransmitted_{};
+  uint64_t packets_retransmitted_{};
+  uint64_t bytes_sent_{};
+  Tracing::Reason trace_reason_;
+  bool health_check_request_{};
   bool should_scheme_match_upstream_{false};
   bool should_drain_connection_{false};
-  OptRef<const StreamInfo> parent_stream_info_;
+  bool is_shadow_{false};
```

Comment:
reordered these field to ensure bool at the end of the object. By this way, even we added a new string, the finally object is no larger than the original one. :)